graph/config/cytoscape/cytoscape.go

```diff
@@ -613,6 +622,14 @@ func generateBoxCompoundNodes(box map[string][]*NodeData, nodes *[]*NodeWrapper,
 
 				// For logical boxing (app), copy some member attributes to to the box node
 				if boxBy == graph.BoxByApp {
+					if nd.HealthData == nil {
+						// make sure to use app health for the app box
+						if graph.IsOK(n.Workload) {
+							nd.HealthData = n.HealthDataApp
+						} else {
+							nd.HealthData = n.HealthData
+						}
```

Comment:
I haven't ran the code, so asking to be sure:
Isn't this logic inverted?

From the old code in the no longer existent `collectGraphHealth` function:
```go
		case graph.NodeTypeApp:
			if workloadOk {
				kind = "workload"
			} else {
				kind = "app"
			}
```

Although I'm not sure if this part should be mirroring that old code.

Reply:
It's not inverted, although I had to stare at it for a moment.  We want the health of the app for the appBox.  versionedApp nodes store both the appHealth (in HealthDataApp), and the workload health (in HealthData).  HealthDataApp is specfically to use for setting the appBox health, whereas HealthData is for the versionedApp node itself (which is really a workload under the covers).

Reply:
Ok, Ok... I think it's also clearer after checking the health appender.

I think that the check is OK, because inside the box we can have several kinds of nodes, but we are only interested in "versionedApp" ones.

Then, perhaps there should be no `else` branch?
I see we can have svc nodes inside the an app box. So, if it happens that the svc node is first on the list, it's health would be assigned to it and then we would have the wrong data on the app box, no?

Again, I still haven't ran the code, so not sure if what I'm saying is just on my imagination :-)

Reply:
> Then, perhaps there should be no else branch?

Well for an app graph (not a versioned app graph) there will be app nodes inside.  So we want to use HealthDataApp for versionedApp nodes, but HealthData for app nodes.

> I see we can have svc nodes inside the an app box. So, if it happens that the svc node is first on the list, it's its health would be assigned to the app box and then we would have the wrong data on the app box, no?

Hmm, I think this is a good catch.  We should only be using app nodes.  Let me push a fix.

Reply:
pushed the change, thanks!